sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/SqlQuerySpecLogger.java

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.models.SqlQuerySpec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SqlQuerySpecLogger {
```

Comment:
`com.azure.cosmos` is a public package of azure-cosmos SDK, do we really need this class as a public surface API ?
@Blackbaud-MikeLueders - Do we plan to use this class in `azure-spring-data-cosmos` SDK ?
If not, I would rather have it in `com.azure.cosmos.implementation` package. 

If for some reason, we need to make it as public class in public surface area, please change it to `final`

Reply:
No reason it needs to be public, class moved.  

Re: making public classes `final`, I would very much appreciate insight into this design decision.  My organization uses a test framework (Spock) that does not currently support mocking final classes which makes unit testing classes that integrate with the Microsoft SDK classes impossible - I basically need to define wrappers around these classes which are just pass-throughs to work around the class itself being final.  For classes which define a public interface, this is not a problem - but not all public classes within these sdk libraries do this.  So I'd very much like to understand the benefit of making public classes final.

Reply:
The reason we make public classes `final` is because that makes them impossible to extend from. That is a design decision which we try to make on every possible new public class. Unless we are sure that the class is fully baked, we keep it `final`. The biggest reason to do this is this: 

Since SDK's implementation keep changing, as we introduce new features, making a class `final` allows us to easily make changes to the instance fields of any `final` class without breaking end user's code. (so basically without introducing breaking changes)

Reply:
@kushagraThapar I had a feeling that was the reason.  As a consumer of libraries, I'd like to offer an alternate perspective.  I would seldom, if ever choose to extend a class from one of these libraries in production code.  By far, the more common scenario would be a test scenario where I may choose to create a mock or stub.  If I did choose to extend a class, I would accept the possibility that the internal implementation would change in a subsequent release and I would need to deal with this.  

To mark a class final takes away options and the stated reason it's being done is to protect me (the consumer) from myself (because a future change might be breaking).  If fields are private, they are already not part of the public interface and should not be expected to never change.  

I realize there's very little chance of this design decision changing but I would like y'all to understand there are consequences to it and (from my perspective at least) the cons are much more impacting than the pro of not breaking code when people rely on internal implementation details.

Reply:
Very valid point @Blackbaud-MikeLueders , however, regarding the mocking / stubbing, we actually suggest our customers to use mockito library - which allows to mock final classes. It is a very common and good library that is being used industry wise. 
I think at this point, we both can agree on the pros / cons of making it final. That being said, we can certainly think of more scenarios (specially this one) - when introducing a public `final` class.